pkg/services/hcloud/server/server.go

```diff
@@ -798,19 +862,22 @@ func (s *Service) createLabels() map[string]string {
 	}
 }
 
-func filterHCloudSSHKeys(sshKeysAPI []*hcloud.SSHKey, sshKeysSpec []infrav1.SSHKey) ([]*hcloud.SSHKey, error) {
-	sshKeysAPIMap := make(map[string]*hcloud.SSHKey)
-	for i, sshKey := range sshKeysAPI {
-		sshKeysAPIMap[sshKey.Name] = sshKeysAPI[i]
+// convertCaphSSHKeysToHcloudSSHKeys converts the ssh keys from the spec to hcloud ssh keys.
```

Comment:
It doesn't convert checks whether there is a match and if that is the case, returns th respective ssh key. Plz adjust the name

Reply:
@janiskemper I updated the comment

// Returns error, if a caphSSHKey is not in allHcloudSSHKeys.

Please tell me your desired name.